### DISCUSSION TITLE
How are the number of steps per epoch calculated?

### DISCUSSION THREAD

**Discussion by jreus** [acids-ircam/RAVE#90] [Q&A] [answered]
Hey everyone. I have a general question about training.
Has anyone who is training models had the experience of radically different steps-per-epoch between different datasets? 

I'm talking about the output line:
`Epoch 4:  75%|███████▍  | 585/782 [05:18<^C:48,  1.83it/s, v_num=1]`

the `585/782` ...which I think is the number of steps (1 step = processing of 1 batch) that we use to figure out when to stop a given stage of RAVE training (by default 1000000 steps for the VAE stage).

...the total number of steps (in this example: 782) seems to be all over the place from model to model for me, sometimes it can be as low as 16, and in this case 782.

I was just curious if anyone could definitively explain how this number of steps per epoch is calculated?

Thanks!

**Comment by lang216** [comment]
@acids-ircam I also found this confusing. It can be very helpful if you can answer! Thanks!

**Reply by lang216** [reply]
I think it has something to do with batch size. However, I still can't get the exact number displayed...

**Comment by caillonantoine** [comment, marked as answer]
1 epoch = the model has "seen" the entirety of the dataset once
there are `dataset_length / batch_size` steps _per_ epoch. One must take into consideration the split between validation and training sets to get the exact numbers.
